File: v2/snippets/csharp/common/objectConfig.cs
using Javonet.Netcore.Sdk;
using Javonet.Netcore.Sdk.Tests.Utils;
using Xunit;
using Xunit.Abstractions;

namespace Javonet.Netcore.Sdk.Tests.common
{
    public class JavaScriptBrowserConfigTests
    {
        private readonly ITestOutputHelper output;
        public BrowserConfigTests(ITestOutputHelper output)
        {
            this.output = output;
        }

        [Fact]
        [Trait("Test", "Integration")]
        public void Test_Channel_WithConfigurationObject_Success()
        {
        // <WithConfigurationObject>
        // Configuration channel object is not yet supported in C#.
        // </WithConfigurationObject>
        }
    }
}

File: v2/snippets/javascript/common/objectConfig.js
const { Javonet } = require("javonet-nodejs-sdk");
const { describe, expect, test } = require("@jest/globals");
const ActivationCredentials = require("../../utils/ActivationCredentials");

describe("Browser config tests", () => {
  test(`Test_Channel_WithConfigurationObject_Success `, () => {
    // <WithConfigurationObject>
    // use Activate only once in your app
    Javonet.activate(ActivationCredentials.yourLicenseKey);

    // Set up configuration Object
    let configObject = {
      runtimes: {
        jvm: [
          {
            name: "default",
            customOptions: "",
            modules: "./",
            channel: {
              type: "inMemory",
            },
          },
        ],
      },
    };

    let communicationChannel = Javonet.withConfig(configObject);
    // use communicationChannel to create runtimes to interact with
    // </WithConfigurationObject>
    expect(communicationChannel).not.toBeNull();
  });
});
